pkgs/top-level/all-packages.nix

```diff
@@ -4589,21 +4589,27 @@ with pkgs;
   cudnnPackages = callPackages ../development/libraries/science/math/cudnn { };
   inherit (cudnnPackages)
     cudnn_7_4_cudatoolkit_10_0
-    cudnn_7_6_cudatoolkit_10_2
+    cudnn_7_6_cudatoolkit_10_0
+    cudnn_7_6_cudatoolkit_10_1
     cudnn_7_6_cudatoolkit_10
+    cudnn_8_1_cudatoolkit_10_2
     cudnn_8_1_cudatoolkit_11_0
     cudnn_8_1_cudatoolkit_11_1
     cudnn_8_1_cudatoolkit_11_2
+    cudnn_8_1_cudatoolkit_10
     cudnn_8_1_cudatoolkit_11
+    cudnn_8_3_cudatoolkit_10_2
     cudnn_8_3_cudatoolkit_11_0
     cudnn_8_3_cudatoolkit_11_1
     cudnn_8_3_cudatoolkit_11_2
     cudnn_8_3_cudatoolkit_11_3
     cudnn_8_3_cudatoolkit_11_4
     cudnn_8_3_cudatoolkit_11_5
+    cudnn_8_3_cudatoolkit_10
     cudnn_8_3_cudatoolkit_11;
 
-  cudnn = cudnn_7_6_cudatoolkit_10;
+  # TODO(samuela): This is old and should be upgraded to 8.3 at some point.
+  cudnn = cudnn_7_6_cudatoolkit_10_1;
```

Comment:
Doesn't this have to be `cudnn_8_3_cudatoolkit_10` in order to be compatible with the currently packaged `cudatoolkit`?

Reply:
Ideally, yes. The issue is that cuDNN 7.6.5 doesn't actually support CUDA 10.2, the current `cudatoolkit` version. And I didn't want to upgrade cuDNN past 7.6.5 here in the interest of keeping this PR as small as possible. The solution is to upgrade `cudnn` to 8.3 which supports CUDA 10.2-11.5. (Looks like you're already on that, great!)

In the future, it think it would be nice to explore a solution like `cudatoolkit_11.withPackages` similar to `python3.withPackages`, but we have to move one step at a time to get the ball rolling.